internal/apis/amko/v1alpha1/types.go

```diff
@@ -154,3 +154,53 @@ type TrafficSplitElem struct {
 type GDPStatus struct {
 	ErrorStatus string `json:"errorStatus,omitempty"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GSLBHostRule is the top-level type which allows a user to override certain
+// fields of a GSLB Service.
+type GSLBHostRule struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// spec for GSLB Config
+	Spec GSLBHostRuleSpec `json:"spec,omitempty"`
+	// +optional
+	Status GSLBHostRuleStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GSLBHostRuleList is a list of GDP resources
+type GSLBHostRuleList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []GSLBHostRule `json:"items"`
+}
+
+// GSLBHostRuleSpec defines all the properties of a GSLB Service that can be overriden
+// by a user.
+type GSLBHostRuleSpec struct {
+	// GSFqdn is the fqdn of the GSLB Service for which the below properties can be
+	// changed.
+	GSFqdn string `json:"gsFqdn,omitempty"`
```

Comment:
It's already a GSLBHostRuleSpec, so shall we just call this field FQDN?

Reply:
FQDN in small caps

Reply:
Yeah, makes sense. Will change.